zerver/tests/test_reminders.py

```diff
@@ -0,0 +1,230 @@
+import datetime
+import time
+from typing import TYPE_CHECKING
+
+import time_machine
+
+from zerver.actions.scheduled_messages import try_deliver_one_scheduled_message
+from zerver.lib.test_classes import ZulipTestCase
+from zerver.lib.timestamp import timestamp_to_datetime
+from zerver.models import Message, ScheduledMessage
+
+if TYPE_CHECKING:
+    from django.test.client import _MonkeyPatchedWSGIResponse as TestHttpResponse
+
+
+class RemindersTest(ZulipTestCase):
+    def do_schedule_reminder(
+        self,
+        message_id: int,
+        scheduled_delivery_timestamp: int,
+    ) -> "TestHttpResponse":
+        self.login("hamlet")
+
+        payload = {
+            "message_id": message_id,
+            "scheduled_delivery_timestamp": scheduled_delivery_timestamp,
+        }
+
+        result = self.client_post("/json/reminders", payload)
+        return result
+
+    def create_reminder(self, content: str, message_type: str = "direct") -> ScheduledMessage:
+        if message_type == "stream":
+            message_id = self.get_stream_message(content)
+        else:
+            message_id = self.get_direct_message(content)
+
+        scheduled_delivery_timestamp = int(time.time() + 86400)
+        result = self.do_schedule_reminder(message_id, scheduled_delivery_timestamp)
+        self.assert_json_success(result)
+        return self.last_scheduled_reminder()
+
+    def last_scheduled_reminder(self) -> ScheduledMessage:
+        return ScheduledMessage.objects.all().order_by("-id")[0]
```

Comment:
Probably this should have a filter to avoid catching non-reminder scheduled messages?

Reply:
Yeah, done.